apps/server/test-i9n/BridgeApp.i9n.ts

```diff
@@ -1,48 +1,217 @@
 import { DynamicModule, Module } from '@nestjs/common';
-import { ConfigModule, registerAs } from '@nestjs/config';
-import { HardhatNetwork, HardhatNetworkContainer, StartedHardhatNetworkContainer } from 'smartcontracts';
+import { ConfigModule } from '@nestjs/config';
+import { ethers } from 'ethers';
+import {
+  BridgeProxy,
+  BridgeProxy__factory,
+  BridgeV1,
+  BridgeV1__factory,
+  EvmContractManager,
+  HardhatNetwork,
+  HardhatNetworkContainer,
+  StartedHardhatNetworkContainer,
+} from 'smartcontracts';
 
+import { AppConfig } from '../src/AppConfig';
 import { AppModule } from '../src/AppModule';
 import { BridgeServerTestingApp } from '../src/BridgeServerTestingApp';
 
 @Module({})
 export class TestingExampleModule {
-  static register(startedHardhatContainer: StartedHardhatNetworkContainer): DynamicModule {
-    const hardhatConfig = registerAs('ethereum', () => ({
-      rpcUrl: startedHardhatContainer.rpcUrl,
-    }));
-
+  static register(config: AppConfig): DynamicModule {
     return {
       module: TestingExampleModule,
-      imports: [AppModule, ConfigModule.forFeature(hardhatConfig)],
+      imports: [AppModule, ConfigModule.forFeature(() => config)],
     };
   }
 }
 
+export function buildTestConfig(startedHardhatContainer: StartedHardhatNetworkContainer, contractAddress?: string) {
```

Comment:
Same as above. We can also do strongly type return 
```suggestion
export function buildTestConfig(startedHardhatContainer: StartedHardhatNetworkContainer, contractAddress?: string) : AppConfig {....
```

Reply:
Another thing, since i'm guessing this test config will only get larger (postgres, etc), with different types, I would prefer that this is as explicit as possible with what users need to pass in. And ideal function call would be

```ts
buildTestConfig({
  hardhatContainer: startedHardhatContainer,
  bridgeProxyAddress: 'some_address',
  ...
})
```
This way it clear (something like named params) on what people need to pass in to build the test config.